baywatch.install

```diff
@@ -8,100 +8,49 @@
+    BaywatchOperationTrait::enable_preview();
 
-  // Enable the new share links module.
-  baywatch_update_8007();
+    // Enable the new share links module.
+    BaywatchOperationTrait::enable_share_links();
 
-  // Revoke the previewer role.
-  baywatch_update_8008();
-}
+    // Remove the Previewer role from users.
+    BaywatchOperationTrait::remove_previewer_role();
 
-/**
- * Import sdpa password policy.
- */
-function baywatch_import_sdpa_password_policy() {
-  $module_installer = \Drupal::service('module_installer');
-  $module_handler = \Drupal::moduleHandler();
-  // Enables required password_policy module.
-  if (!$module_handler->moduleExists('password_policy')) {
-    $module_installer->install(['password_policy']);
-  }
-  // Enables required sub modules.
-  if (!$module_handler->moduleExists('password_policy_character_types')) {
-    $module_installer->install(['password_policy_character_types']);
-  }
-  if (!$module_handler->moduleExists('password_policy_characters')) {
-    $module_installer->install(['password_policy_characters']);
-  }
-  if (!$module_handler->moduleExists('password_policy_consecutive')) {
-    $module_installer->install(['password_policy_consecutive']);
-  }
-  if (!$module_handler->moduleExists('password_policy_history')) {
-    $module_installer->install(['password_policy_history']);
-  }
-  if (!$module_handler->moduleExists('password_policy_length')) {
-    $module_installer->install(['password_policy_length']);
-  }
-  if (!$module_handler->moduleExists('password_policy_username')) {
-    $module_installer->install(['password_policy_username']);
-  }
-  if (!$module_handler->moduleExists('password_strength')) {
-    $module_installer->install(['password_strength']);
-  }
-  // Remove default password policy if exists.
-  $config_factory = \Drupal::configFactory();
-  $config = $config_factory->getEditable('password_policy.password_policy.default');
-  $default_id = $config->get('id');
-  if (!empty($default_id)) {
-    $config->delete();
-    echo "Password policy with id " . $default_id . " has been deleted.\n";
-  }
-  $configs = [
-    'password_policy.password_policy.password_policy_sdpa' => 'password_policy',
-  ];
-  module_load_include('inc', 'tide_core', 'includes/helpers');
-  $config_location = [drupal_get_path('module', 'baywatch') . '/config/optional'];
-  foreach ($configs as $config => $type) {
-    $config_read = _tide_read_config($config, $config_location, TRUE);
-    $storage = \Drupal::entityTypeManager()->getStorage($type);
-    $id = substr($config, strrpos($config, '.') + 1);
-    if ($storage->load($id) == NULL) {
-      $config_entity = $storage->createFromStorageRecord($config_read);
-      $config_entity->save();
-    }
-  }
+    // Remove the Late Runtime Processor module.
+    BaywatchOperationTrait::remove_purge_lateruntime();
 }
 
 /**
  * Import sdpa password policy as an update.
  */
 function baywatch_update_8001() {
-  baywatch_import_sdpa_password_policy();
```

Comment:
Update hooks shouldn't be retrospectively changed. It can lead to unexpected outcomes. As a general rule once an update hook has been pushed to prod it's set in stone. Forget your past mistakes and clean up in a new one.